server/src/main/java/org/opentosca/toscana/plugins/kubernetes/docker/mapper/BaseImageMapper.java

```diff
@@ -66,9 +80,40 @@ Stores the update intervall for the base Image mappings (in hours)
     private MapperEngine engine;
 
     @Autowired
-    public BaseImageMapper(DockerBaseImages[] dockerBaseImages) {
+    public BaseImageMapper(DockerBaseImages[] dockerBaseImages, Preferences preferences) {
         this.baseImages = dockerBaseImages;
         engine = new MapperEngine(imageMap);
+        dataDir = new File(preferences.getDataDir(), DOCKER_IMAGE_DIR);
+        dataDir.mkdirs();
+        imageTagsFile = new File(dataDir, "image_tags");
+        lastUpdateFile = new File(dataDir, "last_update");
+        deserializeTags();
+
+    }
+
+    private void deserializeTags() {
+        try {
+            if (imageTagsFile.exists()) {
+                ObjectInputStream ois = new ObjectInputStream(new FileInputStream(imageTagsFile));
+                Object o = ois.readObject();
```

Comment:
Why not try to wrap the list and the last update value in a object and serialize that using Jackson (ObjectMapper)?

Reply:
Good idea, I'll try.